as-vending/main.go

```diff
@@ -73,21 +73,16 @@ func main() {
 	vendingState.InferenceDataReceived = false
 	vendingState.InferenceWaitThreadStopChannel = inferenceStopChannel
 
-	var err error
-
-	err = service.AddRoute("/boardStatus", vendingState.BoardStatus, "POST")
-	errorAddRouteHandler(lc, err)
-
-	err = service.AddRoute("/resetDoorLock", vendingState.ResetDoorLock, "POST")
-	errorAddRouteHandler(lc, err)
-
-	err = service.AddRoute("/maintenanceMode", vendingState.GetMaintenanceMode, "GET", "OPTIONS")
-	errorAddRouteHandler(lc, err)
-
+	controller := routes.NewController(lc, service, vendingState)
+	err := controller.AddAllRoutes()
+	if err != nil {
+		lc.Errorf("failed to add all Routes: %s", err.Error())
+		os.Exit(1)
+	}
 	// create the function pipeline to run when an event is read on the device channels
 	err = service.SetFunctionsPipeline(
-		transforms.NewFilterFor(vendingState.Configuration.DeviceNames).FilterByDeviceName,
-		vendingState.DeviceHelper,
+		transforms.NewFilterFor(controller.GetVendingState().Configuration.DeviceNames).FilterByDeviceName,
```

Comment:
`GetVendingState` doesn't seem appropriate on the Controller.

Reply:
other option is to expose vending state in controller struct?

Reply:
vending state is already in main for you to use

Reply:
This is the same issue we are seeing in as-controller-board. VendingState is essentially our controller but a different name. One option is to move the VendingState struct as a variable of the controller and change how we are getting the data.